platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java

```diff
@@ -891,6 +909,16 @@ private ImportResponse configureProject(ImportProject importProject, FolderEntry
         return importResponse;
     }
 
+    private boolean resolveProjectType(FolderEntry projectFolder) {
```

Comment:
Don't you think that <code>resolveProjectType</code> is not the best name in this case?
Suggestions: <code>isKnownProjectType</code>, <code>hasProjectType</code>.

Also i think that detection of project type is pretty common issue.
Consider moving _detection logic_ to util class and use it! (if it is not done yet).